src/main/java/com/routeme/api/UserController.java

```diff
@@ -0,0 +1,101 @@
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.routeme.dto.UserDTO;
+import com.routeme.model.DummyRoute;
+import com.routeme.predictionio.PredictionIOClient;
+import com.routeme.service.UserService;
+
+@RestController
+@RequestMapping(value = "/api/users")
+public final class UserController {
+    private final UserService service;
+    private PredictionIOClient predictionIOClient;
+
+    @Autowired
+    public UserController(UserService service) {
+        this.service = service;
+        this.predictionIOClient = new PredictionIOClient();
+    }
+
+    @RequestMapping(method = RequestMethod.GET, produces = "application/json")
+    @ResponseStatus(HttpStatus.OK)
+    List<UserDTO> findAll() {
+        return service.findAll();
+    }
+
+    @RequestMapping(method = RequestMethod.POST, consumes = "application/json", produces = "application/json")
+    @ResponseStatus(HttpStatus.CREATED)
+    UserDTO create(@RequestBody @Valid UserDTO userEntry) {
+        UserDTO userRecord = service.create(userEntry);
+        String userId = userRecord.getId();
+        predictionIOClient.addUserToClient(userId);
+        predictionIOClient.preferRouteType(userId, PredictionIOClient.PREFERENCE_ROUTE_TYPE_LEASTTIME);
+        predictionIOClient.preferRouteMode(userId, PredictionIOClient.PREFERENCE_ROUTE_MODE_BUS);
+        predictionIOClient.preferRouteMode(userId, PredictionIOClient.PREFERENCE_ROUTE_MODE_BUS);
+        predictionIOClient.preferRouteMode(userId, PredictionIOClient.PREFERENCE_ROUTE_MODE_UBAHN);
+        return userRecord;
+    }
+
+    @RequestMapping(value = "{id}/take/{routeId}", method = RequestMethod.GET, produces = "application/json")
+    @ResponseStatus(HttpStatus.CREATED)
+    void takeRoute(@PathVariable("id") String userId, @PathVariable("routeId") String routeId) {
+        predictionIOClient.takeRoute(userId, routeId);
+    }
+
+    @RequestMapping(value = "{id}/view/{routeId}", method = RequestMethod.GET, produces = "application/json")
+    @ResponseStatus(HttpStatus.CREATED)
+    void viewRoute(@PathVariable("id") String userId, @PathVariable("routeId") String routeId) {
+        predictionIOClient.viewRoute(userId, routeId);
+    }
+
+    @RequestMapping(value = "{id}/viewlast/{routeId}", method = RequestMethod.GET, produces = "application/json")
+    @ResponseStatus(HttpStatus.CREATED)
+    void ViewRouteLast(@PathVariable("id") String userId, @PathVariable("routeId") String routeId) {
+        predictionIOClient.viewRouteLast(userId, routeId);
+    }
+
+    @RequestMapping(value = "/{id}", method = RequestMethod.DELETE, produces = "application/json")
+    UserDTO delete(@PathVariable("id") String id) {
+        return service.delete(id);
+    }
+
+    @RequestMapping(value = "/{id}", method = RequestMethod.GET, produces = "application/json")
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://www.codacy.com/app/heshamhamdymassoud/RouteMe-API/file/2580673864/issues/source?bid=3287637&fileBranchId=3302754#l76)